File: 119-pascals-triangle-ii/pascals-triangle-ii.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> row = vector<int>(1, 1);
        for (int i = 1; i <= rowIndex; ++i)
        {
            for (int j = row.size() - 1; j > 0; --j)
            {
                row[j] = row[j] + row[j-1];
            }
            row.push_back(1);
        }
        return row;
    }
};

// time complexity:- O(n^2) -> nested loop.

// space complexity:- O(rowI^2) -> this is because we have n rows in pascals traingle. 
                                // and there are almost n elements in each rows. 
                                // which makes it n^2. 

// need to have a more optimised solution. 

File: 119-pascals-triangle-ii/pascals-triangle-ii.py
class Solution:
    def getRow(self, rowI: int) -> List[int]:
        # declaring a list representing rows of pascals traingle. 

        # 1 element stands for the first row. 
        res = [[1]]
        
        # iterate through rows.
        for i in range(rowI):
            temp = [0] + res[-1] + [0]
            row = []

            # iterate through elements of the row. 
            for j in range(len(res[-1])+1):
                row.append(temp[j] + temp[j+1])
            res.append(row)

        # returns the last row -> rowI. 
        return res[-1]